src/pages/Library.tsx

```diff
@@ -41,24 +53,69 @@ type ILibraryState = {
     model: IModel,
     context: ILibraryContext,
     scrollStamp: number,
-    showVerseOptions: boolean
+    showVerseOptions: boolean,
+    showFab: boolean,   
+    prevHeight: number, 
+    showArrows: boolean, 
+    animateArrows: boolean,
+    animateFab: boolean,
+    last: boolean,
 }
 
 class Library extends React.Component<ILibraryProps, ILibraryState>
 {
+    
+    
     /* Member data */
     private cm : ContentManager;
-    private ref : any;
+    private lut : { [sec_part : string] : string; }; // for article mode next-title (excitingTitle) lookups. sec_part example: "12_1"
 
     constructor(props: ILibraryProps) {
         super(props);
 
         this.cm = this.props.contentManager;
-        this.ref = React.createRef();
+        this.lut = {"0_0":"The Beginning of Knowledge", 
```

Comment:
Hmm. This is fragile. If you want to see how I did it: https://github.com/willie-shen/ProverbsBox/blob/d51bd965f2a64cd3e4dcb17e9f0db84e8aee5571/src/components/AllPopoverContent.tsx#L90

Try this:
        const lut_array = ProverbsStructure.Sections.map<string[][]>((section, i) => {
            let usingPart : boolean = section.Start.Ch !== section.End.Ch;
            if (usingPart) {
                let partCount = section.End.Ch - section.Start.Ch + 1;
                // @ts-ignore
                return ([...Array(partCount).keys()].map<string[]>(part => [i+"_"+part, section.Name + ", pt. " + (part + 1)]));
            }
            else {
                return [[i+"_0", section.Name]];
            }
        }).flat(1)
        this.lut = Object.fromEntries(lut_array);

Reply:
:0 amazing, worked wonderfully!! Thanks for all your help, as usual.